File: app/src/at-linhle/java/intro/TheJourneyBegins.java
package intro;

public class TheJourneyBegins {
    /**
     * Write a function that returns the sum of two numbers.
     */
    int add(int param1, int param2) {
        return param1 + param2;
    }

    /**
     * Given a year, return the century it is in. The first century spans from the year 1 up to
     * and including the year 100, the second - from the year 101 up to and including the year
     * 200, etc.
     */
    int centuryFromYear(int year) {
        int temp = year / 100;
        if (year % 100 == 0) {
            return temp;
        } else return temp + 1;
    }

    /**
     * Given the string, check if it is a palindrome.
     */
    boolean checkPalindrome(String inputString) {
        for (int i = 0; i < ((inputString.length() - 1) / 2F); i++) {
            if (inputString.charAt(i) != inputString.charAt((inputString.length() - 1) - i)) {
                return false;
            }
        }
        return true;
    }
}

File: app/src/at-vuongphan/java/kotlin/TheJourneyBegins.kt
object TheJourneyBegins {
    private const val DIV = 100

    @JvmStatic
    fun main(args: Array<String>) {
        /**
         * Write a function that returns the sum of two numbers.
         */
        val result1 = add(5,6)
        println("1.Result: $result1")
        /**
         * Given a year, return the century it is in. The first century spans
         * from the year 1 up to and including the year 100, the second - from the year 101 up to and including the year 200, etc.
         */
        println("2.Result: ${centuryFromYear(2001)}")
        /**
         *Given the string, check if it is a palindrome.
         */
        println("3.Result: ${checkPalindrome("aabaa")}")

    }
    private fun add(param1: Int, param2: Int): Int {
        return param1 + param2
    }
    private fun centuryFromYear(year: Int): Int {
        if(year % 100 ==0){
            return year / DIV
        }else{
            return year / DIV +1
        }
    }
    private fun checkPalindrome(inputString: String):String{
        return (inputString.reversed()==inputString).toString()
    }
}
